crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -474,16 +485,22 @@ impl<'a> Submitter<'a> {
             tx.clone().access_list(access_list.clone()).estimate_gas()
         )?;
 
+        ensure!(
+            gas_before_access_list > gas_after_access_list,
+            "access list exist but does not lower the gas usage"
+        );
+        let gas_percent_saved = (gas_before_access_list.to_f64_lossy()
+            - gas_after_access_list.to_f64_lossy())
+            / gas_before_access_list.to_f64_lossy()
+            * 100.;
         tracing::debug!(
-            "gas before/after access list: {}/{}, access_list: {:?}",
+            "gas before/after access list: {}/{}, access_list: {:?}, gas percent saved: {}",
             gas_before_access_list,
             gas_after_access_list,
             access_list,
+            gas_percent_saved
         );
-        ensure!(
-            gas_before_access_list > gas_after_access_list,
-            "access list exist but does not lower the gas usage"
-        );
+        track_access_list_gas_saved(gas_percent_saved);
         Ok(access_list)
```

Comment:
nit: I think we should be measuring the gas saved in absolute units and not in percentage.

Reply:
Currently the metric is a histogram. We should probably adjust the buckets either way for the value range we expect. Or this might be a metric where direct logging is more appropriate than prometheus.

Reply:
Ah ok... In that case it makes sense to use a histogram for the % savings, but it might also be interesting in recording the total gas savings as a counter to see how much gas we have saved over time.

Reply:
Instead of a histogram we could have "transaction count" and "gas saved count". Don't think we need to have individual buckets that much. Or you can implement this as a histogram with only one bucket (+inf).

Reply:
Currently we have a histogram but will adjust if we conclude it's needed.